### PR TITLE
Re #3815: Audio refactoring

### PR DESCRIPTION
**Pull request by projectkk2glider**
Should improve (or even fix) #3815  and #1838.

### CODE REVIEW

**Comment by projectkk2glider** [comment]
Although this will compile, it needs the #3877 to properly function (if a file has long file name).

File: radio/src/audio_arm.h [hunk 1]
```diff
@@ -66,16 +66,16 @@ template <unsigned int NUM_BITS> class BitField {
...
 #define AUDIO_SAMPLE_RATE              (32000)
 #define AUDIO_BUFFER_DURATION          (10)
 #define AUDIO_BUFFER_SIZE              (AUDIO_SAMPLE_RATE*AUDIO_BUFFER_DURATION/1000)
 
-#if defined(SIMU_AUDIO)
+#if defined(SIMU) && defined(SIMU_AUDIO)
   #define AUDIO_BUFFER_COUNT           (10) // simulator needs more buffers for smooth audio
 #elif defined(PCBHORUS)
-  #define AUDIO_BUFFER_COUNT           (2)
+  #define AUDIO_BUFFER_COUNT           (5)
```
Comment (bsongis) [inline radio/src/audio_arm.h]: The more we increase the buffer count, the less reactive the vario will be!

**Comment by projectkk2glider** [comment]
Branch rebased and squashed (because of the merge conflicts).

**Comment by bsongis** [comment]
Thanks a lot!

**Pull request merged**